src/core/BufferAttribute.js

```diff
@@ -311,7 +314,11 @@ class BufferAttribute {
 
 	set( value, offset = 0 ) {
 
-		if ( this.normalized ) value = normalize( value, this.array );
+		if ( this.normalized && Array.isArray( value ) ) {
+
+			throw new TypeError( 'THREE.BufferAttribute: array should be a Typed Array.' );
+
+		}
```

Comment:
@gkjohnson this might be a reasonable solution to #24511? I guess my priorities are:

1. be sure we can give set(...) the same argument type as the constructor
2. try not to break backward compatibility any more than necessary

This throws the same error as the constructor. I think it's probably better than converting values in some cases and not others.

Reply:
Sounds fine to me!